src/main/java/me/snowdrop/fallback/AnnotationAwareFallbackOperationsInterceptor.java

```diff
@@ -49,65 +51,46 @@ public AnnotationAwareFallbackOperationsInterceptor(BeanFactory beanFactory) {
 
 	@Override
 	public Object invoke(MethodInvocation invocation) throws Throwable {
-		final MethodInterceptor delegate = getDelegate(invocation.getThis(), invocation.getMethod());
-		if (delegate != null) {
-			return delegate.invoke(invocation);
+		final MethodInterceptor methodInterceptor = getMethodInterceptor(invocation.getThis(), invocation.getMethod());
+		if (methodInterceptor != null) {
+			return methodInterceptor.invoke(invocation);
 		}
 		else {
 			return invocation.proceed();
 		}
 	}
 
-	private MethodInterceptor getDelegate(Object target, Method method) {
-		if (!this.delegatesCache.containsKey(target) || !this.delegatesCache.get(target).containsKey(method)) {
-			synchronized (this.delegatesCache) {
-				if (!this.delegatesCache.containsKey(target)) {
-					this.delegatesCache.put(target, new HashMap<>());
+	private MethodInterceptor getMethodInterceptor(Object target, Method method) {
+		if (!this.methodInterceptorCache.containsKey(target) || !this.methodInterceptorCache.get(target).containsKey(method)) {
+			synchronized (this.methodInterceptorCache) {
+				if (!this.methodInterceptorCache.containsKey(target)) {
```

Comment:
You can change this to `putIfAbsent` so all the if + create is reduced to one line.